services/QuillLMS/client/app/bundles/PremiumHub/components/dataExportTableAndFields.tsx

```diff
@@ -166,7 +173,11 @@ export const DataExportTableAndFields = ({ queryKey, searchCount, selectedGrades
       grades: selectedGrades,
       headers_to_display: getHeaders().map(header => header.attribute)
     }
+    setDownloadButtonBusy(true)
+
     requestPost('/snapshots/create_csv_report_download', requestParams, (body) => {
+      setShowSnackbar(true)
+      setTimeout(() => {setDownloadButtonBusy(false)}, buttonDisableTime);
```

Comment:
I recall that previous React versions needed to explicitly clear timers. I notice in our code we don't do that. Perhaps the latest version of React handles timer cleanup, or it's just not relevant for short-lived components?

Reply:
Good question. React does not handle clearing timeouts on its own. My understanding is that clearing timeouts is important so that you don't have weird side effects that can occur if the timeout gets called after the component has unmounted. I generally only find myself using them in pretty high-level components (such that we are less likely to "unmount" rather than just navigate away, like this page), so for practical purposes it doesn't matter, but I think there's an argument that we should still be explicitly clearing the timeout just in case.